parsec/api/protocol/handshake.py

```diff
@@ -355,12 +354,16 @@ def build_result_req(self, verify_key: Optional[VerifyKey] = None) -> bytes:
                 answer = self.answer_data["answer"]
                 assert isinstance(answer, bytes)
 
-                if self.client_api_version >= (2, 5):
+                # Provides compatibility with API version 2.4 and below
+                # TODO: Remove once API v2.x is deprecated
+                if self.client_api_version.version == 2 and self.client_api_version.revision < 5:
```

Comment:
why not `if (2, 0) <= self.client_api_version <= (2, 5)` ?

Reply:
btw this works because we don't support apiv1 authenticated handshake anymore, but in theory this function should 1) check api version (and typically set boolean flags used by the rest of the function) 2) do the `if self.answer_type == HandshakeType.AUTHENTICATED:`

Reply:
> why not if (2, 0) <= self.client_api_version <= (2, 5) 

Sure, why not.

> btw this works because we don't support apiv1 authenticated handshake anymore, but in theory this function should 1) check api version (and typically set boolean flags used by the rest of the function) 2) do the if self.answer_type == HandshakeType.AUTHENTICATED:

I typically like to keep this compatibility as local as possible. Do you mean something like:
```python
use_old_signing_scheme = (2, 0) <= self.client_api_version <= (2, 5)
[...]
if self.answer_type == HandshakeType.AUTHENTICATED:
    [...]
    if use_old_signing_scheme:
        [...]
    else:
        [...]
```
or am I missing something? :thinking: